pkg/prometheus/server/rules.go

```diff
@@ -110,7 +109,12 @@ func (c *Operator) createOrUpdateRuleConfigMaps(ctx context.Context, p *monitori
 		return currentConfigMapNames, nil
 	}
 
-	newConfigMaps, err := makeRulesConfigMaps(p, newRules)
+	newConfigMaps, err := makeRulesConfigMaps(
+		p,
+		newRules,
+		operator.WithAnnotations(c.config.Annotations),
+		operator.WithLabels(c.config.Labels),
```

Comment:
before this change, the operator-defined labels and annotations weren't propagated to the rules configmaps.